src/index.ts

```diff
@@ -60,44 +148,23 @@ document.getElementById('clear-button')!.onclick = function() {
     redraw();
 };
 
-const parameterSlider = document.getElementById('parameter-slider') as HTMLInputElement;
-parameterSlider.max = (fftSize - 1).toString();
-parameter = parameterSlider.valueAsNumber;
-parameterSlider.oninput = function() {
-    parameter = parameterSlider.valueAsNumber;
-    redraw();
-};
-const complexityNumber = document.getElementById('complexity-number') as HTMLInputElement;
-complexityNumber.max = (fftSize - 1).toString();
-complexity = complexityNumber.valueAsNumber;
-complexityNumber.oninput = function() {
-    complexity = complexityNumber.valueAsNumber;
-    redraw();
-};
-const complexityCircles = document.getElementById('complexity-circles-check') as HTMLInputElement;
-circles = complexityCircles.checked;
-complexityCircles.oninput = function() {
-    circles = complexityCircles.checked;
-    redraw();
-};
+document.getElementById('save-button')!.onclick = setLocation;
 
 function magnitude(x: number, y: number) { return Math.sqrt(x * x + y * y); }
 
 function lerp(first: number, second: number, t: number) { return first + (second - first) * t; }
 
-function addPoint(x: number, y: number) {
+function addPoint(x: number, y: number, draw = true) {
     if (points.length === 0) {
         points.push({ x, y, segmentLength: 0 });
+        points.push({ x, y, segmentLength: 0 });
     } else {
-        const previousPoint = points[Math.max(0, points.length - 2)];
+        const previousPoint = points[points.length - 2];
         const segmentLength = magnitude(x - previousPoint.x, y - previousPoint.y);
         unclosedLength += segmentLength;
-
-        const addedPoint = { x, y, segmentLength };
-        points.splice(points.length - 1, 0, addedPoint);
-
-        const startAndEndPoint = points[points.length - 1];
-        startAndEndPoint.segmentLength = magnitude(startAndEndPoint.x - x, startAndEndPoint.y - y);
+        points[points.length - 1] = { x, y, segmentLength };
+        const firstPoint = points[0];
+        points.push({ x: firstPoint.x, y: firstPoint.y, segmentLength: magnitude(firstPoint.x - x, firstPoint.y - y) });

```

Comment:
Could you do this without changing how this works? This is reverting newer code that I think I prefer.
It would then not be necessary to skip any points. If you prefer, you could move the last point to become first when you save, but there is really an equivalence since it's a loop.

Reply:
I did some wrong copy-past, didn't I?
I think that was part of the d025c9b reapplication…

Reply:
138ef2b - sorry for the mispasted code.

Reply:
No worries! Thanks for all the work you're doing!